server/server_test.go

```diff
@@ -423,6 +422,14 @@ p = true { false }`
 			// Subsequent query should not fail.
 			tr{http.MethodGet, "/query?q=x=1", "", 200, `{"result": [{"x": 1}]}`},
 		}},
+		{"delete and check", []tr{
+			tr{http.MethodDelete, "/data/a/b", "", 404, ""},
+			tr{http.MethodPut, "/data/a/b/c/d", `1`, 204, ""},
+			tr{http.MethodGet, "/data/a/b/c", "", 200, `{"result": {"d": 1}}`},
+			tr{http.MethodDelete, "/data/a/b", "", 204, ""},
+			tr{http.MethodGet, "/data/a/b/c/d", "", 200, ""},
```

Comment:
If the test result response is `""` the body won't be checked. I think we want to check that the result is an empty object in this case: `"{}"`.